File: app/src/main/java/com/example/zyyschedule/viewmodel/MainActivityViewModel.java
package com.example.zyyschedule.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.zyyschedule.database.DataRepository;
import com.example.zyyschedule.database.Schedule;

import java.util.List;

public class MainActivityViewModel extends AndroidViewModel {
    private final DataRepository dataRepository;

    public MainActivityViewModel(@NonNull Application application) {
        super(application);
        this.dataRepository = new DataRepository(application);
    }
    public LiveData<List<Schedule>> getALLUnFinishOfRemind(){
        return dataRepository.getALLUnFinishOfRemind();
    }

    public void updateRemindTag(Integer ...id){
        dataRepository.updateRemindTag(id);
    }
}

File: app/src/main/java/com/example/zyyschedule/viewmodel/MainActivityViewModel.kt
package com.example.zyyschedule.viewmodel

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.example.zyyschedule.database.DataRepository
import com.example.zyyschedule.database.Label
import com.example.zyyschedule.database.Schedule
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainActivityViewModel(application: Application) : AndroidViewModel(application) {
    private val dataRepository: DataRepository = DataRepository(application)

    fun getALLUnFinishOfRemind(): LiveData<List<Schedule>> {
        return dataRepository.getALLUnFinishOfRemind()
    }

    fun updateRemindTag(vararg id: Int) {
        viewModelScope.launch(Dispatchers.IO) {
            try {
                dataRepository.updateRemindTag(*id)
            } catch (e: Exception) {
                Log.i("main", "更改提醒状态失败：$e")
            }
        }
    }

    fun deleteSchedule(vararg schedules: Schedule) {
        viewModelScope.launch(Dispatchers.IO) {
            try {
                dataRepository.deleteSchedule(*schedules)
            } catch (e: Exception) {
                Log.i("main", "删除日程失败：$e")
            }
        }
    }

    fun getLabelTitle(id: Int): LiveData<Label> {
        return dataRepository.getLabelTitle(id)
    }

}